pyface/image/tests/test_image.py

```diff
@@ -0,0 +1,582 @@
+# (C) Copyright 2005-2021 Enthought, Inc., Austin, TX
+# All rights reserved.
+#
+# This software is provided without warranty under the terms of the BSD
+# license included in LICENSE.txt and may be redistributed only under
+# the conditions described in the aforementioned license. The license
+# is also available online at http://www.enthought.com/licenses/BSD.txt
+#
+# Thanks for using Enthought open source!
+
+from contextlib import closing
+from os import stat
+from importlib_resources import files
+from pathlib import Path
+import shutil
+import tempfile
+import time
+import unittest
+from zipfile import ZipFile, ZIP_DEFLATED
+
+from pyface.image_resource import ImageResource
+from pyface.ui_traits import Border, Margin
+from ..image import (
+    FastZipFile, ImageLibrary, ImageVolume, ImageVolumeInfo, ZipFileReference,
+    join_image_name, split_image_name, time_stamp_for,
+)
+
+
+LIBRARY_DIR = files('pyface.image') / "library"
+ICONS_FILE = LIBRARY_DIR / "icons.zip"
+TEST_IMAGES_DIR = files('pyface.tests') / "images"
+
+
+class TestJoinImageName(unittest.TestCase):
+
+    def test_simple(self):
+        image_name = join_image_name("icons", "red_ball.jpg")
+
+        self.assertEqual(image_name, "@icons:red_ball.jpg")
+
+    def test_extension(self):
+        image_name = join_image_name("icons", "red_ball.png")
+
+        self.assertEqual(image_name, "@icons:red_ball")
+
+    def test_double_extension(self):
+        image_name = join_image_name("icons", "red_ball.foo.png")
+
+        self.assertEqual(image_name, "@icons:red_ball.foo.png")
+
+
+class TestSplitImageName(unittest.TestCase):
+
+    def test_simple(self):
+        volume_name, file_name = split_image_name("@icons:red_ball.jpg")
+
+        self.assertEqual(volume_name, "icons")
+        self.assertEqual(file_name, "red_ball.jpg")
+
+    def test_extension(self):
+        volume_name, file_name = split_image_name("@icons:red_ball")
+
+        self.assertEqual(volume_name, "icons")
+        self.assertEqual(file_name, "red_ball.png")
+
+    def test_no_at_symbol(self):
+        volume_name, file_name = split_image_name("icons:red_ball.jpg")
+
+        # XXX this should probably raise rather than doing this
```

Comment:
Do you think it makes sense to open an issue for this instead of leaving this information hidden in a comment in a test?

Reply:
Done.